regressao-linear/regressao-linear.md

```diff
@@ -0,0 +1,91 @@
+
+## Ideia principal
+
+Imagine que você possui uma série de pontos em um gráfico, esses pontos formam o nosso conjunto de dados de aprendizado, a ideia por trás da regressão linear é encontrar a reta que mais se aproxima da distribuição dos nossos pontos. Exemplificando:
+
+![](./imagens/exemplo-reta.png)
+
+Aparentemente, a reta se ajusta bem à distribuição de dados, o que é verdade, mas precisamos primeiro definir a métrica que nos permite comparar duas retas e dizer qual delas melhor se "ajusta" à distribuição de dados. Imagine que nossa reta genérica é de formato $$ f(x) = ax+b $$ e nosso conjunto de dados é formado por pontos $$(x_i,y_i)$$, poderíamos adotar como métrica simplesmente a soma do erro para cada ponto, o que nos daria:
+
+$$Erro = f(x_0) - y_0 + f(x_1) - y_1 + ...$$
+$$Erro = \sum_{i = 0}^{N}f(x_i) - y_i$$
+
+Todavia, note que se nosso conjunto de dados fosse formado por $$(1,2)$$ e $$(2,-2)$$, e também nossa reta tivesse como resultados $$f(1) = 0$$ e $$f(2)=0$$, nosso erro final seria:
+
+$$Erro = (0 - 2) + (0 - (-2)) = -2 + 2 = 0$$
+
+Ou seja, essa métrica acaba não sendo muito útil devido ao fato dela abrir margem para que um erro possa anular outro e nos impedir de perceber que uma reta candidata é uma má reta. Desse modo, podemos adicionar um quadrado nos termos do somatório, desse modo todos os erros ficam positivos e temos uma noção melhor do que está acontecendo, chamamos esse erro de erro quadrático . Assim sendo:
+
+$$Erro = \sum_{i=0}^{N} (f(x_i) - y_i)^2 $$
+
+Porém, imagine que sua quantidade de pontos é muito grande. Em um computador, talvez uma soma de quadrados tão grande possa acarretar em overflow ou acarretar no acúmulo de erros de precisão. Desse modo, é interessante que, ao invés de utilizarmos o erro quadrático simples, seja utilizado o erro quadrático médio - *Mean Squared Error (MSE)*:
+
+$$Erro = \frac{\sum_{i=0}^{N} (f(x_i) - y_i)^2}{N}$$
+
+Com a nossa métrica de erro estabelecida, finalmente podemos comparar retas e formalizar nosso objetivo: encontrar uma reta que minimiza o erro quadrático médio.
+
+## Quando o erro é mínimo
+
+Na matéria de cálculo é aprendido o que é o gradiente de uma função, um conceito essencial não só para a regressão linear, mas também para a área de aprendizado de máquina como um todo, especialmente no que se refere a redes neurais. Simplificando, o gradiente de uma função indica a direção e a taxa de variação máxima de uma função em um determinado ponto. Desse modo, quando o gradiente de uma função é zero em um determinado ponto, isso significa que aquele ponto é crítico, isso é: ou aquele ponto é máximo (local ou global), ou é mínimo (local ou global), ou é um ponto de sela. Como nossa função de erro é estritamente positiva, então sabemos que o ponto onde o gradiente dela é zero é o erro mínimo que podemos chegar.
```

Comment:
- Não sei se mencionar Redes Neurais aqui é muito bom onde estamos num capítulo onde não introduzimos o conceito de redes neurais ... Talvez só dizer que é bem útil pra ML já basta?
- "Simplificando, o gradiente de uma função indica a **direção** e a taxa de variação ..." Direção de quê? Poderia deixar um pouco mais claro esse trecho?
- "quando o gradiente de uma função é zero em um determinado ponto, isso significa que aquele ponto é crítico, isso é: ou aquele ponto é máximo (local ou global), ou é mínimo (local ou global), ou é um ponto de sela" isso somente é verdade para funções de R^2 pra R.

Reply:
- Creio que mencionar redes neurais não machuca ninguém, por isso voto em deixar. Posso adicionar um "que é um assunto que veremos no futuro" ou algo assim se for do seu agrado.
- Pra pessoa entender o gradiente descendente ela precisa ter visto cálculo antes, creio que um leitor com esse contato tenha ideia do que é gradiente e pontos críticos, e por isso entenda o que quero dizer. Mas, posso dizer que é a direção que a função cresce mais rápido, te agrada?
- Me estranha essa sua afirmação pq, na minha cabeça, isso invalida o uso de gradiente descendente para outras configurações de funções. Se for do seu agrado, e vc estiver certo, adiciono essa informação de que isso é específico para funções de R^2 pra R.

Reply:
- @jooag o que vc acha? Acho que mencionar "que é um assunto que veremos no futuro" pode ser também
- Acho uma boa
- Não invalida não! O problema na afirmação está em dizer que o ponto vai ser um ponto de sela se não for máximo ou mínimo local da função. Porém, pontos de selas são definidos para superfícies, que caracterizam uma função em R^2. Dito isso, acho que podemos especificar que isso é válido pra R^2 e dizemos que pode ser, no geral, generalizável pra R^n com suas respectivas estruturas análogas.

Reply:
- Adicionei o 'assunto que veremos no futuro'
- Adicionado
- Adicionei esse R^2 em R

Mudanças poderão ser vistas a partir do commit 'Alterações para o Fernando - Parte II'